src/components/SetupOpportunity/SetupOpportunity.tsx

```diff
@@ -53,18 +84,28 @@ export const SetupOpportunity: FC<Props> = ({
     opportunity,
     updateWebsiteListingRanking,
     updateApplicationRanking,
-    loading,
+    finishOpportunity,
+    loading
 }) => {
-    const [allOpportunities, setAllOpportunities] = useState();
-    const linkToFunders = opportunity.getOpportunity
-        ? `/setup/${opportunity.getOpportunity.id}/funders`
+    const { getOpportunity } = opportunity;
+    const [allWorkflows, setAllWorkflows] = useState<
+        (ApplicationListing | WebsiteListing | null)[]
+    >([]);
+    const onButtonClick = useCallback(() => {
+        finishOpportunity();
+    }, [finishOpportunity]);
+
+    const linkToFunders = getOpportunity
+        ? `/setup/${getOpportunity.id}/funders`
         : "";
 
     useEffect(() => {
-        if (!loading) {
-            setAllOpportunities(getAllOpportunities(opportunity));
+        if (!getOpportunity) {
+            return;
         }
-    }, [loading]);
+
+        setAllWorkflows(getAllWorkflows(getOpportunity));
+    }, [getOpportunity]);
```

Comment:
This may result in the flicker effect again.

Reply:
Ah, that must have crept in with the merge

Reply:
Fixed